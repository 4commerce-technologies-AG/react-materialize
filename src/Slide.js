import React from 'react';
import PropTypes from 'prop-types';

const Slide = ({ src, children }) => {
  if (typeof src === 'string') {
    return (
      <li>
        <img src={src} alt="" />
        {children}
      </li>
    );
  } else {
    return (
      <li>
        {src}
        {children}
      </li>
    );
  }
};

Slide.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  /**
   * The path of the background image
   */
  src: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired
};

export default Slide;
