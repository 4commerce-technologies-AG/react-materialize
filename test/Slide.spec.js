import React from 'react';
import { shallow } from 'enzyme';
import Slide from '../src/Slide';

describe('<Slide />', () => {
  let wrapper = shallow(<Slide image="#" />);

  test('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render a Slide', () => {
    expect(wrapper.childAt(0).is('img')).toEqual(true);
    expect(wrapper.childAt(0).is({ src: '#' })).toEqual(true);
  });

  test('can render a picture element', () => {
    const pictureTag = (
      <picture>
        <source srcSet="#" type="image/webp" />
        <source srcSet="#" type="image/jpg" />
        <img src="#" alt="" />
      </picture>
    );

    wrapper = shallow(<Slide image={pictureTag} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('can render a custom caption', () => {
    wrapper = shallow(
      <Slide image="#">
        <h2>test</h2>
      </Slide>
    );
    expect(wrapper.childAt(1).is('h2')).toEqual(true);
  });
});
