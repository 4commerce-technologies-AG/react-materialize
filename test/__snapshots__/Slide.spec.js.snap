// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slide /> can render a picture element 1`] = `
<li>
  <picture>
    <source
      srcSet="#"
      type="image/webp"
    />
    <source
      srcSet="#"
      type="image/jpg"
    />
    <img
      alt=""
      src="#"
    />
  </picture>
</li>
`;

exports[`<Slide /> renders 1`] = `
<li>
  <img
    src="#"
  />
</li>
`;
